### ISSUE TITLE
Custom Configuration of fzf and Help for rga-fzf and rga-fzf-open

### ISSUE THREAD

**Issue by lamyergeier** [phiresky/ripgrep-all#207]
Please include help for  rga-fzf and rga-fzf-open (like  `rga-fzf -h`,  `rga-fzf-open -h`)

I just started using this tool, and I am not sure how to use these. I have fzf. May be prevent fzf  from using its using its own config file as I get the following error: Error: fzf output not two line

May we can specify the config from fzf  in `ripgrep-all/config.jsoc` in the same way `mpv` specify the config for `yt-dlp` so that it does not conflicts with the standard config of `yt-dlp` and can override it.

**Comment by juanMarinero** [comment]
I also want to customize [rga-fzf](https://github.com/phiresky/ripgrep-all/wiki/fzf-Integration).

`xdg-open` is fine, but I want to vim to the **line** found in regex match.

How to edit [rga-fzf](https://github.com/phiresky/ripgrep-all/wiki/fzf-Integration) so I get both filename and line of regex match? And if file of type text (and default text editor is vim, alike: `xdg-mime query default text/plain`) then vim that file to the returned line.

Workaround **without** _rpa_. Thanks to @BuonOmo [comment](https://github.com/sharkdp/bat/issues/1185#issuecomment-1301473901). Call it like:
`fzf_jump_to_regex_in_multiple_files "regex1" "regex2" "etc"`
Press `Enter` to vim file at regex match line.

```
fzf_jump_to_regex_in_multiple_files() {   
  # ripgrep (rg) searchs an empty regex pattern (everything) --> every line of every file --> slow !!
  rg --line-number --no-heading "" |
	fzf \
	--delimiter=: \
	--preview='batcat --color=always {1} --highlight-line={2}' \
	--preview-window='~3,+{2}+3/4' \
        --query="$*" \
        --bind "enter:become(vim {1} +{2})" \
        --bind 'down:preview-down,up:preview-up'
} 
```

**Comment by juanMarinero** [comment]
Amend. To search not only in CWD, but in every file/dir passed.
```sh
fzf_jump_to_regex_in_multiple_files_passed() {
  # Call example to search regex in 1 file and 1 dir:
  # fzf_jump_to_regex_in_multiple_files_passed \
  #  --files ~/.vimrc ~/.oh-my-zsh --regex "regex1 regex2"
  
  local search_paths=()
  local query=""

  while [[ $# -gt 0 ]]; do
    case "$1" in
      --files) shift ;;
      --regex) query="$2"; shift 2 ;;    # Set regex explicitly
      *) search_paths+=("$1"); shift ;;  # Treat as path
    esac
  done

  # If no paths given, default to current dir (.)
  [[ ${#search_paths[@]} -eq 0 ]] && search_paths=(".")

  rg --follow --line-number --no-heading --hidden "" "${search_paths[@]}" |
    sed "s|^$HOME|~|" |  # Replace "$HOME" with "~" for cleaner display
    fzf \
      --delimiter=: \
      --preview='batcat --color=always "$(echo {1} | sed "s|^~|$HOME|")" --highlight-line={2}' \
      --preview-window='~3,+{2}+3/4' \
      --query="$query" \
      --bind "enter:become(vim \"\$(echo {1} | sed 's|^~|$HOME|')\" +{2})" \
      --bind 'down:preview-down,up:preview-up'
}
```

Note. `--files` flag is optional. Next two commands are equivalent
```sh
fzf_jump_to_regex_in_multiple_files_passed --files file.py --regex "regex1 regex2"
fzf_jump_to_regex_in_multiple_files_passed         file.py --regex "regex1 regex2"
```

We can further customize for fixed files/dirs. Example:
```sh
fzf_jump_to_regex_vim_editor() {
  # Call example:
  # fzf_jump_to_regex_vim_editor regex1 regex2

  fzf_jump_to_regex_vim_editor \
    --files ~/.vimrc ~/.vim/ftplugin ~/.vim/plugin \
    --regex "${*:-}"
}
```

Actually to just search in CWD one can overwrite `fzf_jump_to_regex_in_multiple_files` (of previous comment) with next because CWD is the default search-path (if non passed).
```sh
fzf_jump_to_regex_in_multiple_files() {
  # Call example:
  # fzf_jump_to_regex_in_multiple_files regex1 regex2

  fzf_jump_to_regex_in_multiple_files_passed --regex "${*:-}"
}
```
General note: I believe I made a poor choose of function names. Maybe `fzf_jump_to_regex_in_multiple_files` could be renamed to `fzf_jump_to_regex_in_CWD` and `fzf_jump_to_regex_in_multiple_files_passed` just `fzf_jump_to_regex_in_multiple_files`.

**Comment by juanMarinero** [comment]
Amend. Vim open multiple files at same lines as shown in FZF.

```sh
...
fzf \
      --multi \
      --delimiter=: \
      ...
      --bind "Ctrl-t:become(vim_cmd=\$(echo {+2} | tr ' ' '\n' | awk 'NR==1{print \$1} NR>1{print \"tabnext | \"\$1}'); ${EDITOR:-vim} -p {+1} -c \"\$vim_cmd\")"
```

**TL;DR**
1. Add `--multi \` and `--bind "Ctrl-t:...` lines above to my [previous solution](https://github.com/phiresky/ripgrep-all/issues/207#issuecomment-2772448020)
2. Source that script. And run `fzf_jump_to_regex_in_multiple_files_passed`
3. [Optional] type a regex.
4. Select multiple matches with `TAB`
5. Trigger `Ctrl+T` so it opens in `vim` (or your `$EDITOR`) each match in a new buffer with cursor in the respective line as in FZF was shown. Note, same file can be opened in several buffers, each with its cursor-line.

**Example**

![Image](https://github.com/user-attachments/assets/c77d4944-3cde-413e-b03b-44b389189237)

**Explanation**


1. Guides
 - https://github.com/junegunn/fzf/blob/master/README.md#using-the-finder:
> On multi-select mode (-m), TAB and Shift-TAB to mark multiple items
 - https://thevaluable.dev/practical-guide-fzf-example/#previews-placeholder
2. Code

 - `--bind "Ctrl-t:become(echo {+})"`
  Echoes in `--multi` e.g. `./a.txt:1:line1content ./b.txt:2:line2content` which has next syntax per file selected `path:number:regex<space>`. But `regex` and `path` can have spaces, thus avoid using `sed` on `{+}`, use instead `{+1}` to get space list of files and `{+2}` to get space list of line numbers

 - `--bind "Ctrl-t:become(echo {+1})"`
  Echoes in `--multi` e.g. `./a.txt ./b.txt ./c.txt`
  
 - `--bind "Ctrl-t:become(echo {+2})"`
  Echoes the line numbers in `--multi`, e.g.: `3 2 69`

 - Vim how to open multiple files per tab with custom line number:
   -  To open multiple files in vim next syntax is **not** valid: `vim -p a.txt +3 b.txt +2 c.txt +69`
   -  Instead one must use alike `vim -p a.txt b.txt c.txt -c '3 | tabnext | 2 | tabnext | 69'`
   -  Alternative `vim -p a.txt b.txt -c '/regex'` using `--bind "Ctrl-t:become(${EDITOR:-vim} -p {+1} -c /{q})"`

 - `--bind "Ctrl-t:become(echo {+2} | tr ' ' '\n' | awk 'NR==1{print \$1} NR>1{print \"tabnext | \"\$1}')"`
  Echoes e.g.  `3 | tabnext | 2 | tabnext | 69`
